cai_causal_graph/causal_graph.py

```diff
@@ -1569,6 +1569,24 @@ def get_parents_graph(self, node: NodeLike) -> CausalGraph:
 
         return parents_graph
 
+    def _filter_graph_by_nodes(self, nodes: List[NodeLike]) -> CausalGraph:
+        """
+        Get a subgraph consisting only of nodes in list and edges between those nodes.
+        """
+        node_list: List[str] = [self._NodeCls.identifier_from(node) for node in nodes]
+        filtered_edges = [
+            edge
+            for edge in self.edges
+            if edge.source.identifier in node_list and edge.destination.identifier in node_list
+        ]
+        filtered_graph = CausalGraph()
+        if len(filtered_edges) == 0:
+            for node in nodes:
+                filtered_graph.add_node(self.get_node(node))
```

Comment:
Easier to just do this:

```python
if len(filtered_edges) == 0:
    filtered_graph.add_nodes_from(node_list)
```

Reply:
Actually, you're probably doing this to keep the meta of each node, right? In that case ignore what I said :)